framework/py/flwr/serverapp/strategy.py

```diff
@@ -162,88 +162,121 @@ def start(
-            Optional function for centralized evaluation. Takes server round number
-            and array record, returns a MetricRecord. If provided, will be called
-            before the first round and after each round. Defaults to None.
+        global_evaluate_fn : Callable[[int, ArrayRecord], MetricRecord], optional
+            Optional function for centralized evaluation of the global model. Takes
+            server round number and array record, returns a MetricRecord. If provided,
+            will be called before the first round and after each round. Defaults to
+            None.
 
         Returns
         -------
-        StrategyResults
-            Results containing training metrics, evaluation metrics, centralized
-            evaluation metrics (if provided), and final model arrays from all rounds.
+        Results
+            Results containing final model arrays and also training metrics, evaluation
+            metrics and global evaluation metrics (if provided) from all rounds.
         """
         log(INFO, "Starting %s strategy:", self.__class__.__name__)
         log_strategy_start_info(
             num_rounds, initial_arrays, train_config, evaluate_config
         )
-        self.summary()
-        log(INFO, "")
+        summary = self.summary()
+        log(INFO, "%s\n", summary)
 
         # Initialize if None
         train_config = ConfigRecord() if train_config is None else train_config
         evaluate_config = ConfigRecord() if evaluate_config is None else evaluate_config
-        results = Result()
+        result = Result()
 
         t_start = time.time()
-        # Do central eval with starting global parameters
-        if central_eval_fn:
-            res = central_eval_fn(0, initial_arrays)
-            log(INFO, "Initial central evaluation results: %s", res)
-            results.central_evaluate_metrics[0] = res
+        # Evaluate starting global parameters
+        if global_evaluate_fn:
+            res = global_evaluate_fn(0, initial_arrays)
+            log(INFO, "Initial global evaluation results: %s", res)
+            result.global_evaluate_metrics[0] = res
 
         arrays = initial_arrays
         for current_round in range(1, num_rounds + 1):
             log(INFO, "")
             log(INFO, "[ROUND %s/%s]", current_round, num_rounds)
 
-            # Configure train, send messages and wait for replies
-            replies = grid.send_and_receive(
-                messages=self.configure_train(
-                    current_round, arrays, train_config, grid
-                ),
+            # -----------------------------------------------------------------
+            # --- TRAINING ----------------------------------------------------
+            # -----------------------------------------------------------------
+
+            # Call strategy to configure training round
+            train_messages = self.configure_train(
```

Comment:
In the previous design we purposely made the `self.config_train(...)` call directly as an input arg to `send_and_receive()` so to avoid having 2xN models in memory (N sent, then N received).